package/MDAnalysis/lib/util.py

```diff
@@ -212,6 +214,182 @@ def filename(name, ext=None, keep=False):
     return name if isstream(name) else str(name)
 
 
+def get_reader_for(filename, format=None):
```

Comment:
I would rather have `get_*_for()`  in `core`; it just feels out of place in `lib`.

Having the registries at the top level seems appropriate.